lib/ansible/modules/cloud/azure/azure_rm_aks.py

```diff
@@ -0,0 +1,594 @@
+            service_principal=dict(
+                type='dict',
+                options=service_principal_spec
+            ),
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+        self.dns_prefix = None
+        self.kubernetes_version = None
+        self.tags = None
+        self.state = None
+        self.linux_profile = None
+        self.agent_pool_profiles = None
+        self.service_principal = None
+
+        required_if = [
+            ('state', 'present', [
+             'dns_prefix', 'linux_profile', 'agent_pool_profiles', 'service_principal'])
+        ]
+
+        self.results = dict(changed=False, state=dict())
+
+        super(AzureRMManagedCluster, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                    supports_check_mode=True,
+                                                    supports_tags=True,
+                                                    required_if=required_if)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        resource_group = None
+        to_be_updated = False
+
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
+            self.location = resource_group.location
+
+        response = self.get_aks()
+
+        # Check if the AKS instance already present in the RG
+        if self.state == 'present':
+            # For now Agent Pool cannot be more than 1, just remove this part in the future if it change
+            agentpoolcount = len(self.agent_pool_profiles)
+            if agentpoolcount > 1:
+                self.fail(
+                    'You cannot specify more than one agent_pool_profiles currently')
+
+            self.results['state'] = response
+            if not response:
+                to_be_updated = True
+
+            else:
+                self.log('Results : {0}'.format(response))
+                update_tags, response['tags'] = self.update_tags(
+                    response['tags'])
+
+                if response['provisioning_state'] == "Succeeded":
+                    if update_tags:
+                        to_be_updated = True
+
+                    def is_property_changed(profile, property, ignore_case=False):
+                        base = response[profile].get(property)
+                        new = getattr(self, profile).get(property)
+                        if ignore_case:
+                            return base.lower() != new.lower()
+                        else:
+                            return base != new
+
+                    # Cannot Update the SSH Key for now // Let service to handle it
+                    if is_property_changed('linux_profile', 'ssh_key'):
+                        self.log(("Linux Profile Diff SSH, Was {0} / Now {1}"
```

Comment:
My understanding is, which value can be updated is the logic handled by service side. So here show the difference, and let service to decide whether this is updatable.

Reply:
Ok, the comment above this is what confused me, maybe state, the service will determine if the key is updated or not rather than the point saying Cannot Update the SSH Key for now.